vector/src/main/java/im/vector/app/core/epoxy/ExpandableTextItem.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2020 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.core.epoxy
+
+import android.animation.ObjectAnimator
+import android.text.TextUtils
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.core.view.doOnPreDraw
+import androidx.core.view.isVisible
+import com.airbnb.epoxy.EpoxyAttribute
+import com.airbnb.epoxy.EpoxyModelClass
+import im.vector.app.R
+import im.vector.app.core.extensions.copyOnLongClick
+
+@EpoxyModelClass(layout = R.layout.item_expandable_textview)
+abstract class ExpandableTextItem : VectorEpoxyModel<ExpandableTextItem.Holder>() {
+
+    @EpoxyAttribute
+    lateinit var content: String
+
+    @EpoxyAttribute
+    var maxLines: Int = 3
+
+    private var isExpanded = false
+    private var expandedLines = 0
+
+    override fun bind(holder: Holder) {
+        super.bind(holder)
+        holder.content.text = content
+        holder.content.copyOnLongClick()
+
+        holder.content.doOnPreDraw {
+            if (holder.content.lineCount > maxLines) {
+                expandedLines = holder.content.lineCount
+                holder.content.maxLines = maxLines
+
+                holder.view.setOnClickListener {
+                    if (isExpanded) {
+                        collapse(holder)
+                    } else {
+                        expand(holder)
+                    }
+                }
+            } else {
+                holder.arrow.isVisible = false
```

Comment:
if this Item would be recycled, the arrow could be Gone. You should make it visible in the `if` block

Reply:
Done.